.github/actions/generate-metadata/action.yml

```diff
@@ -0,0 +1,55 @@
+---
+name: Build environment
+description: Create build environment
+
+inputs:
+    gh_token:
+        description: GitHub token
+        required: true
+        type: string
+    cache:
+        description: use caching or not
+        required: true
+        type: boolean
+
+runs:
+    using: composite
+    steps:
+        - name: Restore cached Plugins
+          if: inputs.cache
+          id: cache-plugins-restore
+          uses: actions/cache/restore@v3
+          with:
+            path: plugins_metadata.json
+            key: plugins_metadata
```

Comment:
Thanks @ltalirz. This is not working as expected somehow, for the "preview" action, the cache is set the false and this step still runs and cause generate steps skipped. Do you have any idea why?

Reply:
Sorry, I don't know the details of how the if conditions work

Reply:
No worries, I think I find how to exit this rabbit hole. All because of this syntax issue of GitHub action https://github.com/actions/runner/issues/1483 🥲